### ISSUE TITLE
Incorrect version requirement for scipy

### ISSUE THREAD

**Issue by cdiener** [biocore/scikit-bio#1818]
`setup.py` specifies a requirement of scipy>=1.3, however scikit-bio imports some classes that are absent in this scipy version. For instance, with scipy 1.4.1 skbio can not be imported:

```python
ImportError                               Traceback (most recent call last)

[<ipython-input-47-1df0fe07cbc5>](https://localhost:8080/#) in <module>()
----> 1 import skbio
      2 
      3 skbio.__version__

5 frames

[/usr/local/lib/python3.7/dist-packages/skbio/stats/distance/_mantel.py](https://localhost:8080/#) in <module>()
     16 from scipy.stats import PearsonRConstantInputWarning
     17 from scipy.stats import PearsonRNearConstantInputWarning
---> 18 from scipy.stats import SpearmanRConstantInputWarning
     19 
     20 from skbio.stats.distance import DistanceMatrix

ImportError: cannot import name 'SpearmanRConstantInputWarning' from 'scipy.stats' (/usr/local/lib/python3.7/dist-packages/scipy/stats/__init__.py)
```

**Comment by lrvdijk** [comment]
Experiencing the same, SciPy 1.9 removed a couple things:

https://docs.scipy.org/doc/scipy/release.1.9.0.html#backwards-incompatible-changes

**Comment by maxibor** [comment]
Same here :(

**Comment by ebolyen** [comment]
Thanks for the report! If anyone happens to have bandwidth, a PR would be appreciated, but if not, we'll correct this before our next release.

**Comment by eray-sahin** [comment]
Same issue here. How long may it take to correct?

**Comment by ebolyen** [comment]
A release is probably around October. Your best bet at the moment is to downgrade Scipy. If you're good with a source install, then patching this is probably pretty easy since it will just be changing some imports which catch a deprecated error class. 

Our tests ought to already be aware of how to generate these errors, so I think it should be straight-forward to find the new error and swap the implementation slightly.

**Comment by iBuitron** [comment]
Same error:

(venv) root@LPT-2022-035:/home/reveur/python/test# python -m skbio.test
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/home/reveur/python/test/venv/lib/python3.10/site-packages/skbio/__init__.py", line 11, in <module>
    import skbio.io  # noqa
  File "/home/reveur/python/test/venv/lib/python3.10/site-packages/skbio/io/__init__.py", line 247, in <module>
    import_module('skbio.io.format.lsmat')
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/reveur/python/test/venv/lib/python3.10/site-packages/skbio/io/format/lsmat.py", line 77, in <module>
    from skbio.stats.distance import DissimilarityMatrix, DistanceMatrix
  File "/home/reveur/python/test/venv/lib/python3.10/site-packages/skbio/stats/distance/__init__.py", line 197, in <module>
    from ._mantel import mantel, pwmantel
  File "/home/reveur/python/test/venv/lib/python3.10/site-packages/skbio/stats/distance/_mantel.py", line 16, in <module>
    from scipy.stats import PearsonRConstantInputWarning
ImportError: cannot import name 'PearsonRConstantInputWarning' from 'scipy.stats' (/home/reveur/python/test/venv/lib/python3.10/site-packages/scipy/stats/__init__.py)

WSL2 - Ubuntu 22.04 LTS, fresh install.

**Comment by eray-sahin** [comment]
Mine was solved after I removed the current verison and installed the version 1.8.1 via

```
pip3 uninstall scipy
pip3 install scipy=1.8.1
```

Hope it works until next release.

Best,

**Comment by SilasK** [comment]
Should be fixed with #1821

**Comment by ebolyen** [comment]
Fixed by 0.5.8.
